test/powershell/Language/Scripting/HashtableToPSCustomObjectConversion.Tests.ps1

```diff
@@ -0,0 +1,248 @@
+                                                          
+                            public int a;
+                            public int b;
+                        }
+"@
+                 add-type -typedefinition $source
+                 $x = [SampleClass2]@{blah=10;b=5 }
+                 Throw "Exception expected, execution should not have reached here"
+       }
+       catch
+       {
+           $_.Exception.InnerException.ErrorRecord.FullyQualifiedErrorId | Should Be 'ObjectCreationError'
+       }       
+    }
+
+
+    It 'Creating an object of an existing type from hashtable should throw error when setting incompatible values for properties' {       
+       
+       try
+       { 
+          $source = @"
+                        public class SampleClass3
+                        {
+                            public SampleClass3(int x)
+                            {
+                            	a = x;
+                            }
+                            
+                            public SampleClass3()
+                            {
+                            }
+                                                          
+                            public int a;
+                            public int b;
+                        }
+"@
+                 add-type -typedefinition $source
+                 $x = [SampleClass3]@{a="foo";b=5}
+                 Throw "Exception expected, execution should not have reached here"
+       }
+       catch
+       {           
+           $_.Exception.InnerException.ErrorRecord.FullyQualifiedErrorId | Should Be 'ObjectCreationError'
+       }
+    }
+
+    #known issue 2069 
+    It 'Creating an object of an existing type from hashtable should call the constructor taking a hashtable if such a constructor exists in the type' -skip:$IsCoreCLR {
+       {
+                $source = @"
+                        public class SampleClass5
+                        {                                                       
+                            public SampleClass5(int x)
+                            {
+                            	a = x;
+                            }
+                            
+                            public SampleClass5(System.Collections.Hashtable h)   
+                            {
+		                          a = 100;
+		                          b = 200;
+                            }
+                            
+                            public SampleClass5()
+                            {
+                            }
+                                                          
+                            public int a;
+                            public int b;
+                        }
+"@
+                 add-type -typedefinition $source
+                 $script:x = [SampleClass5]@{a=10;b=5}
+            
+       } | Should Not Throw
+       
+       $script:x | Should Not Be $null
+       ($script:x.a) | Should Be '100'
+    }

```

Comment:
fixed.

---
In reply to: [78256443](https://github.com/Powershell/Powershell/pull/2160#discussion_r78256443) [](ancestors = 78256443)